scripts/beck/st/error_injection.py

```diff
@@ -164,128 +176,156 @@ def ReadMetaData(offset):
 
 def induceCorruption(recordType, noOfCorr):
-	print(recordType)
-	print("No of Error want to induce: {}".format(noOfCorr))
+	count = 0
+	read_metadata_file()
+	logger.info(recordType)
+	logger.info("No of Error want to induce: {}".format(noOfCorr))
 	lookupList = recordDict[recordType]
 	if (len(lookupList) & noOfCorr) == 0:
-		print("Record List is empty. Please choose another Record")
+		logger.error("Record List is empty. Please choose another Record")
+		count = 0
+		return count
 	elif len(lookupList) < noOfCorr:
-		print(" Record List contains Less no of entry than Input. Please reduce the number of Error Injection")
+		logger.error(" Record List contains Less no of entry than Input. Please reduce the number of Error Injection")
+		count = 0
+		return count
 	else:
-		print(lookupList)
-		print("**** Inducing {} Error in Record: {} ****".format(noOfCorr,recordType))
+		logger.info(lookupList)
+		logger.info("**** Inducing {} Error in Record: {} ****".format(noOfCorr,recordType))
 		for i in range(noOfCorr):
-			offset = lookupList[i]
+			offset = lookupList[i]  ## Please add offset here for starting from middle of offset list
 			ReadMetaData(offset+8)
-			EditMetadata(offset+8)			
+			EditMetadata(offset+8)
+			count = count + 1
+	return count
 
 def induceRandomCorruption(noOfCorr):
-	recType=random.choice(list(recordDict))
-	print("++++ Picked a Random Record from Dictionary Record type:{}++++".format(recType))
-	print("No of Error want to induce: {}".format(noOfCorr))
-	lookupList = recordDict[recType]
-	print(lookupList)
-	if len(lookupList) == 0:
-		print("Record List is empty. Please choose another Record")
-	elif len(lookupList) < noOfCorr:
-		print(" Record List contains Less no of entry than Input. Please reduce the number of Error Injection")
-	else:
-		lookupList = random.sample(lookupList,noOfCorr)
-		print(lookupList)
-		for i in range(noOfCorr):
-			offset = lookupList[i]
-			print("**** Inducing RANDOM Error in Record at offsets {}****".format(hex(offset+8)))
-			ReadMetaData(offset+8)
-			EditMetadata(offset+8)
-			ReadMetaData(offset+8)
-			print(recordDict)
+	count = 0
+	read_metadata_file()
+	while(1):
+		recType=random.choice(list(recordDict))
+		logger.info("++++ Picked a Random Record from Dictionary Record type:{}++++".format(recType))
+		logger.info("No of Error want to induce: {}".format(noOfCorr))
+		lookupList = recordDict[recType]
+		logger.info(lookupList)
+		if (len(lookupList) == 0) or (len(lookupList) < noOfCorr) :
+			logger.info("Record List is empty OR contains Less no of entry than Input. Going to next Record")
+		else:
+			lookupList = random.sample(lookupList,noOfCorr)
+			logger.info(lookupList)
+			for i in range(noOfCorr):
+				offset = lookupList[i]
+				logger.info("**** Inducing RANDOM Error in Record at offsets {}****".format(hex(offset+8)))
+				ReadMetaData(offset+8)
+				EditMetadata(offset+8)
+				ReadMetaData(offset+8)
+				count = count + 1
```

Comment:
Can we call induceCorruption()?
Otherwise, code are duplicated.

Reply:
Induce Corruption functionality is different. Which is induce corruption at a particular record type. 
No. this is for taking a log like Read, Corrupt, Read at a particular offset and log it. So log will be user friendly for reading.